mountpoint-s3/src/main.rs

```diff
@@ -490,6 +491,27 @@ fn parse_perm_bits(perm_bit_str: &str) -> Result<u16, anyhow::Error> {
     }
 }
 
+/// Validate a bucket name. This isn't intended to be an exhaustive validation, just a quick filter
+/// to catch common CLI mistakes like using an S3 URI (`s3://bucket/`) or a path (`~/mnt`).
+fn parse_bucket_name(bucket_name: &str) -> anyhow::Result<String> {
+    if bucket_name.len() < 3 || bucket_name.len() > 255 {
+        return Err(anyhow!("bucket names must be 3-255 characters long"));
+    }
+
+    if bucket_name.contains("s3://") {
+        return Err(anyhow!("bucket name should not be an s3:// URI (provide the bare bucket name instead; use --prefix for prefix mounts)"));
+    }
+
+    // Actual bucket names must start/end with a letter, but bucket aliases can end with numbers
+    // (-s3), so let's just naively check for invalid characters.
+    let bucket_regex = Regex::new(r"^[0-9a-zA-Z\-\.]+$").unwrap();
+    if !bucket_regex.is_match(bucket_name) {
+        return Err(anyhow!("bucket names can only contain letters, numbers, . and -"));
+    }
```

Comment:
_us-east-1_ allowed underscores in bucket names until 2018, so we should allow that in the regular expression. https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html

(I assume that they can only be used when using path-style addressing.)